src/main/java/edu/ucsb/cs56/ucsb_courses_search/controller/CourseController.java

```diff
@@ -64,6 +65,22 @@ public String index(Model model, OAuth2AuthenticationToken token) throws AccessF
         }
         return "courseschedule/index";
     }
+
+    /**
+     * Deletes all courses with the same name as "name" from "courseList" and CourseRepository.
+     * Returns a list without any courses of name "name".
+    */
+    private List<ScheduleItem> deleteByClassname(List<ScheduleItem> courseList, String name){
```

Comment:
What is this function doing in the controller? Controllers are only for mappings. If you want to do something like this, it should probably be bundled into a service. Also, I'm not sure I see the reason for duplicating the list, when you could just modify the courseList passed in to the function.

Reply:
had trouble adding the service (still looking into it), also i just forgot about passing by reference when i wrote that